src/lib/MedStickerEncryptionV2.js

```diff
@@ -0,0 +1,85 @@
+import create from "./factory";
+import { arrayBufferToString, encode } from "./utilities";
+import { CHARLIE } from "./constants";
+
+const scrypt = create("scrypt");
+const gcm = create("AES-GCM");
+
+/**
+ *
+ * @param secret
+ * @param salt
+ * @returns {PromiseLike<CryptoKeyPair> | PromiseLike<CryptoKey> | PromiseLike<CryptoKeyPair | CryptoKey>}
```

Comment:
Looking at the implementation, this return type is way off what it really will be. Can you check what it really is and use the correct type here. 

Infact, the implementation of the scrypt call in our factory is incorrect:
https://github.com/VivyTeam/krypt-web/blob/master/src/lib/factory.js#L212
https://github.com/dchest/scrypt-async-js#scryptpassword-salt-options-callback

It's only working right now because no one has passed a non-zero value of `interruptStep` to it. 
In the factory its treated as always being a sync call. As soon as someone passes an interrupt step its going to return null.

Reply:
do you recommend that in the factory this should be implemented as an `async`?

Reply:
Either generateKey needs to return a promise, or you need to override the option to always be 0 no matter what the user inputs. 

Look at the signature of the other generateKeys to decide the best option

Reply:
can we please check this together on what the returned value type should be @Weetbix ?